lib/cmds/merge_cmds/show.ts

```diff
@@ -79,27 +77,44 @@ export const getChangesetAndTargetContentType = async ({
     additionalParameters: {
       spaceId: activeSpaceId,
       environmentId: targetEnvironmentId
-    }
+    },
+    timeout
   })
 
-  const [targetContentType, appActionResult] = await Promise.all([
+  const [targetContentType, appActionResult] = await Promise.allSettled([
     ContentTypeApiHelper.getAll({
       client,
       environmentId: targetEnvironmentId,
       spaceId: activeSpaceId
     }),
     appActionCall
   ])
+  if (
+    targetContentType.status === 'rejected' ||
+    appActionResult.status === 'rejected'
+  ) {
+    throw new Error(
+      `${errorEmoji} There was an error generating the migration. Please try again.`
+    )
+  }
 
-  const { result } = appActionResult
+  const { result } = appActionResult.value
 
   if (isResultWithError(result)) {
+    if (result.errorMessage === 'PollTimeout') {
+      throw new Error(
+        `${errorEmoji} The migration took too long to generate. Please try again.`
```

Comment:
Since this message is used in multiple places I wonder if we should store in centrally somewhere where it can be used for any PollTimeout error